src/erlcloud_aws.erl

```diff
@@ -409,9 +424,54 @@ signing_key(Config, Date, Region, Service) ->
     KRegion = erlcloud_util:sha256_mac( KDate, Region),
     KService = erlcloud_util:sha256_mac( KRegion, Service),
     erlcloud_util:sha256_mac( KService, "aws4_request").
-    
+
 authorization(Config, CredentialScope, SignedHeaders, Signature) ->
     ["AWS4-HMAC-SHA256"
      " Credential=", Config#aws_config.access_key_id, $/, CredentialScope, $,,
      " SignedHeaders=", SignedHeaders, $,,
      " Signature=", Signature].
+
+%% This function fetches http://status.aws.amazon.com/data.json
+%% and examine "current" section for on going AWS issues/failures.
+%% Example of a return status:
+%% [{<<"service_name">>,
+%%   <<"Amazon Elastic Compute Cloud (Frankfurt)">>},
+%%  {<<"summary">>,<<"[RESOLVED] Internet Connectivity ">>},
+%%  {<<"date">>,<<"1436972949">>},
+%%  {<<"status">>,1},
+%%  {<<"details">>,<<>>},
+%%  {<<"description">>,
+%%   <<"<div><span class=\"yellowfg\"> 8:22 AM PDT</span>&nbsp;Between 7:55 AM PDT and 8:05 AM PDT we experienced Internet connectivity issues for some instances in the EU-CENTRAL-1 Region. The issue has been resolved and the service is operating normally.</div>">>},
+%%  {<<"service">>,<<"ec2-eu-central-1">>}]
+%%
+%% <<"status">> field values are the following:
+%%  0 - service is operating normally;
+%%  1 - performance issues;
+%%  2 - service disruption.
+-spec get_service_status(list(string())) -> ok | list().
+get_service_status(ServiceNames) when is_list(ServiceNames) ->
+    {ok, Json} = aws_request_form(get, "http", "status.aws.amazon.com", undefined, 
+        "/data.json", "", [], default_config()),
+
+    case get_filtered_statuses(ServiceNames, 
+            proplists:get_value(<<"current">>, jsx:decode(Json)))
+    of
+        [] -> ok;
+        ReturnStatuses -> ReturnStatuses
+    end.
+
+get_filtered_statuses(ServiceNames, Statuses) ->
+    lists:filter(
+        fun(S)-> 
+            lists:any(
+                fun(InputService)-> 
+                    ServiceNameBin = list_to_binary(InputService),
+                    ServiceNameLen = byte_size(ServiceNameBin),
+                    case proplists:get_value(<<"service">>, S) of 
+                        <<ServiceNameBin:ServiceNameLen/binary, _/binary>> -> true;
+                        _ -> false
+                    end
+                end, 
+            ServiceNames)
```

Comment:
please fix indent - `ServiceNames` and `Statuses` should match corresponding `fun`